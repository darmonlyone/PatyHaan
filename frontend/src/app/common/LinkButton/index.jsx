import React from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router";

const LinkButton = ({
	history,
	to,
	onClick,
	className,
	children,
	disabled,
}) => {
	return (
		<button
			disabled={disabled}
			className={className}
			onClick={(event) => {
				onClick && onClick(event);
				history.push(to);
			}}
		>
			{children}
		</button>
	);
};

LinkButton.propTypes = {
	children: PropTypes.node.isRequired,
};

export default withRouter(LinkButton);
