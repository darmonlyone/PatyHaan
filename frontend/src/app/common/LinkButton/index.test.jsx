import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import {
	BrowserRouter as Router,
} from "react-router-dom";

import LinkButton from ".";

let container = null;
beforeEach(() => {
	container = document.createElement("div");
	document.body.appendChild(container);
});

afterEach(() => {
	unmountComponentAtNode(container);
	container.remove();
	container = null;
});

it("should renders LinkButton", () => {
	act(() => {
		render(<Router><LinkButton >test</LinkButton></Router>, container);
	});
	expect(pretty(container.innerHTML)).toMatchSnapshot();

	act(() => {
		render(<Router><LinkButton className="bg-red-500">test</LinkButton></Router>, container);
	});
	expect(pretty(container.innerHTML)).toMatchSnapshot();

	act(() => {
		render(<Router><LinkButton to="/">test</LinkButton></Router>, container);
	});

	expect(pretty(container.innerHTML)).toMatchSnapshot();

	act(() => {
		render(<Router><LinkButton disabled>test</LinkButton></Router>, container);
	});
	expect(pretty(container.innerHTML)).toMatchSnapshot();
});
