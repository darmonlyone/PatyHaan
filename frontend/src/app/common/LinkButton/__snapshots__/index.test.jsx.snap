// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders LinkButton 1`] = `"<button>test</button>"`;

exports[`should renders LinkButton 2`] = `"<button class=\\"bg-red-500\\">test</button>"`;

exports[`should renders LinkButton 3`] = `"<button>test</button>"`;

exports[`should renders LinkButton 4`] = `"<button disabled=\\"\\">test</button>"`;
